### PR TITLE
Migrate 'core settings' e2e tests to Playwright

### PR DESCRIPTION
**Pull request by fai-sal**
# What

Part of #38851.

PR migrates `core-settings.test.js` e2e tests to Playwright.

# Why
- #38851 

# Testing Instructions
`npm run test:e2e:playwright -- test/e2e/specs/editor/various/core-settings.spec.js`

### CODE REVIEW

**Comment by Mamaduka** [comment]
Thanks for contributing, @fai-sal!

Unfortunately, I cannot pull down this branch to test locally. Can make sure you're fork and this branch is up to date - https://developer.wordpress.org/block-editor/contributors/code/git-workflow/#keeping-your-fork-up-to-date?

P.S. I would recommend creating a new file for migration instead of moving the test file and then deleting the old test in a separate comment. This makes it more manageable to deal with merge conflicts.

**Comment by fai-sal** [comment]
Hi @Mamaduka, Thanks for the feedback!. I have synced the branch with trunk, Can you please retry pulling this branch to test locally. 

If still no luck then I can create another PR following your recommended approach.

File: test/e2e/specs/editor/various/core-settings.spec.js [hunk 1]
```diff
@@ -16,22 +16,33 @@ async function getOptionsValues( selector ) {
...
+			page
+		);
 
-		await visitAdminPage( 'options-general.php' );
+		await admin.visitAdminPage( 'options-general.php' );
 		await page.type(
 			'input#blogdescription',
 			'Just another Gutenberg site'
 		);
 		await page.click( 'input#submit' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/core-settings.spec.js]:
```suggestion
		await page
			.getByRole( 'textbox', { name: 'Tagline' } )
			.fill( 'Just another Gutenberg site' );
		await page.getByRole( 'button', { name: 'Save Changes' } ).click();
```

Role-based selectors should be preferred when possible. See https://playwright.dev/docs/locators#quick-guide.

File: test/e2e/specs/editor/various/core-settings.spec.js [hunk 2]
```diff
@@ -16,22 +16,33 @@ async function getOptionsValues( selector ) {
...
 		// We won't select the option that we updated and will also remove some
 		// _transient options that seem to change at every update.
 		const optionsInputsSelector =
 			'form#all-options table.form-table input:not([id*="_transient"]):not([id="blogdescription"])';
-		const optionsBefore = await getOptionsValues( optionsInputsSelector );
+		const optionsBefore = await getOptionsValues(
+			optionsInputsSelector,
+			admin,
+			page
+		);
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/core-settings.spec.js]: Note: I'm not sure we can improve this helper. Maybe @kevin940726 has any suggestions.

File: test/e2e/specs/editor/various/core-settings.spec.js (hunk 1 above)
Comment (fai-sal) [reply test/e2e/specs/editor/various/core-settings.spec.js]: Thanks for the suggestion @Mamaduka . Old Selectors are now updated by role-based selectors as suggested.

Comment (Mamaduka) [reply test/e2e/specs/editor/various/core-settings.spec.js]: Thank you, @fai-sal!

**Review by Mamaduka — APPROVED** [review]
Let's merge this.

Thanks again, @fai-sal!

**Pull request merged**